### PR TITLE
Dreambooth training

### PR DESCRIPTION
**Pull request by entrpn**
- Adds dreambooth training
- Refactors unit tests with new config settings.
- Load sdxl checkpoints from gcs to save disk space.

### CODE REVIEW

File: src/maxdiffusion/maxdiffusion_utils.py [hunk 1]
```diff
@@ -162,21 +172,72 @@ def get_dummy_unet_inputs(config, pipeline):
     time_ids_channels = pipeline.unet.projection_class_embeddings_input_dim - text_embeds_dim
     time_ids_dims = time_ids_channels // pipeline.unet.addition_time_embed_dim
     added_cond_kwargs = {
-      "text_embeds": jnp.zeros((batch_size, text_embeds_dim), dtype=dtype),
-      "time_ids": jnp.zeros((batch_size, time_ids_dims), dtype=dtype),
+      "text_embeds": jnp.zeros((batch_size, text_embeds_dim), dtype=config.weights_dtype),
```
Comment (suexu1025) [inline src/maxdiffusion/maxdiffusion_utils.py:175]: should all those in activation_dtype instead of weight type  or extra input dtype?

Comment (entrpn) [reply src/maxdiffusion/maxdiffusion_utils.py:175]: @suexu1025 I kept them as the same type as weights for simplicity and not to add another dtype in the config. Let me know what you think.

Comment (suexu1025) [reply src/maxdiffusion/maxdiffusion_utils.py:175]: How about we just pass it as fp32, as we only provide fp32 input data to be less error prone?

**Review by suexu1025 — COMMENTED** [review]
Thanks, others look good!

**Review by suexu1025 — APPROVED** [review]

**Pull request merged**